src/ast/ddl.rs

```diff
@@ -270,6 +270,29 @@ pub enum TableConstraint {
         /// Referred column identifier list.
         columns: Vec<Ident>,
     },
+    /// MySQLs [fulltext][1] definition. Since the [`SPATIAL`][2] definition is exactly the same,
+    /// and MySQL displays both the same way, it is part of this definition as well.
+    ///
+    /// Supported syntax:
+    ///
+    /// ```markdown
+    /// {FULLTEXT | SPATIAL} [INDEX | KEY] [index_name] (key_part,...)
+    ///
+    /// key_part: col_name
+    /// ```
+    ///
+    /// [1]: https://dev.mysql.com/doc/refman/8.0/en/fulltext-natural-language.html
+    FulltextOrSpatial {
+        /// Whether this is a `FULLTEXT` (true) or `SPATIAL` (false) definition.
+        fulltext: bool,
+        /// Whether the type is followed by the keyword `KEY` (Some(true)), `INDEX` (Some(false)),
+        /// or no keyword at all (None).
+        opt_key: Option<bool>,
```

Comment:
Maybe an enum would be easier to use here

```rust
enum SpatialKey {
  Key,
  Index,
  None,
}
```

🤔

Reply:
Will do it asap!

Reply:
@alamb done!